openpype/hosts/max/api/lib.py

```diff
@@ -322,18 +323,87 @@ def is_headless():
 
 
 @contextlib.contextmanager
-def viewport_camera(camera):
+def viewport_setup_updated(camera):
+    """Function to set viewport camera during context
+    ***For 3dsMax 2024+
+    Args:
+        camera (str): viewport camera for review render
+    """
     original = rt.viewport.getCamera()
+    has_autoplay = rt.preferences.playPreviewWhenDone
     if not original:
         # if there is no original camera
         # use the current camera as original
         original = rt.getNodeByName(camera)
     review_camera = rt.getNodeByName(camera)
     try:
         rt.viewport.setCamera(review_camera)
+        rt.preferences.playPreviewWhenDone = False
         yield
     finally:
         rt.viewport.setCamera(original)
+        rt.preferences.playPreviewWhenDone = has_autoplay
+
+
+@contextlib.contextmanager
+def viewport_setup(instance, viewport_setting, camera):
+    """Function to set camera and other viewport options
+    during context
+    ****For Max Version < 2024
+
+    Args:
+        instance (str): instance
+        viewport_setting (str): active viewport setting
+        camera (str): viewport camera
+
+    """
+    original = rt.viewport.getCamera()
+    has_vp_btn = rt.ViewportButtonMgr.EnableButtons
+    has_autoplay = rt.preferences.playPreviewWhenDone
+    if not original:
+        # if there is no original camera
+        # use the current camera as original
+        original = rt.getNodeByName(camera)
+    review_camera = rt.getNodeByName(camera)
+
+    current_visualStyle = viewport_setting.VisualStyleMode
+    current_visualPreset = viewport_setting.ViewportPreset
+    current_useTexture = viewport_setting.UseTextureEnabled
+    orig_vp_grid = rt.viewport.getGridVisibility(1)
+    orig_vp_bkg = rt.viewport.IsSolidBackgroundColorMode()
+
+    visualStyle = instance.data.get("visualStyleMode")
+    viewportPreset = instance.data.get("viewportPreset")
+    useTexture = instance.data.get("vpTexture")
+    has_grid_viewport = instance.data.get("dspGrid")
+    bkg_color_viewport = instance.data.get("dspBkg")
```

Comment:
Wouldn't it make more sense to have this context manager explicitly take these parameters - instead of taking from the instance just so the context manager itself is clearer and also better suited for `lib`?